Tests/SentryTests/SentryHubTests.swift

```diff
@@ -715,6 +715,45 @@ class SentryHubTests: XCTestCase {
         XCTAssertEqual(envelope, fixture.client.captureEnvelopeInvocations.first)
     }
 
+    func testCaptureEnvelope_WithUnhandledException() {
+          sut.startSession()
+
+          let beginSession = sut.session
+
+          let event = TestData.event
+          event.level = .error
+          event.exceptions = [TestData.exception]
+          event.exceptions?.first?.mechanism?.handled = false
+          sut.capture(SentryEnvelope(event: event))
+
+          let endSession = sut.session
+          XCTAssertNotEqual(beginSession, endSession)
```

Comment:
`m`: I think we don't need this check. We should be able to do all validations with the envelope below. Then we also don't need to expose the session from the hub.

Reply:
The envelope will contain the old session, we need to check whether we started a new one.

Reply:
You already change the date of the `fixture.currentDateProvider.setDate(date: Date(timeIntervalSince1970: 2))` after starting the session. As you check the timestamp below with `XCTAssertEqual(json["timestamp"] as? String, "1970-01-01T00:00:02.000Z")` I think you don't need this check. It already validates that the hub started a new session cause you validate the timestamp.